istioctl/cmd/wait_test.go

```diff
@@ -74,41 +78,51 @@ func TestWaitCmd(t *testing.T) {
 		},
 		{
 			execClientConfig: cannedResponseMap,
-			args:             strings.Split("x wait --revision canary virtualservice foo.default", " "),
+			args:             strings.Split("x wait --timeout 2s --revision canary virtualservice foo.default", " "),
 			wantException:    false,
 		},
 	}
 
-	_ = setupK8Sfake()
-
 	for i, c := range cases {
 		t.Run(fmt.Sprintf("case %d %s", i, strings.Join(c.args, " ")), func(t *testing.T) {
+			_ = setupK8Sfake()
 			verifyExecTestOutput(t, c)
 		})
 	}
 }
 
 func setupK8Sfake() *fake.FakeDynamicClient {
-	objs := []runtime.Object{
-		newUnstructured("networking.istio.io/v1alpha3", "virtualservice", "default", "foo", "1"),
-		newUnstructured("networking.istio.io/v1alpha3", "virtualservice", "default", "bar", "3"),
-	}
-	client := fake.NewSimpleDynamicClient(runtime.NewScheme(), objs...)
+	client := fake.NewSimpleDynamicClient(runtime.NewScheme())
 	clientGetter = func(_, _ string) (dynamic.Interface, error) {
```

Comment:
This is not safe, will result in data race

Reply:
Can you elaborate?  Is creating a fake client racy?  Is the call to NewScheme racy?

Reply:
writing to clientGetter. I didn't guess this myself, I just ran the test

Reply:
I haven't changed that line in this PR, and it's been consistently passing both in CI and on my system.  Are you running with some special config?